### PR TITLE
docs: Clarity Adoption Guide

### PR DESCRIPTION
**Pull request by bdryanovski**
### Addoption Guide

Small angular application to provide examples on how to adopt Core components and replace existing Angular ones.

**⚠️To trick yarn to keep few packages inside the `node_modules` - we need to link then with the relative path I'm going with a minor fix version - adding them into `nohoist` is trigger more issues into other builds - the packages are `prismjs` and `@webcomponents/webcomponentsjs`**

```
$ cd apps/clarity-adoption
$ yarn run start
```

Code examples are located here `apps/clarity-adoption/src/app/migrations`

### CODE REVIEW

**Comment by gnomeontherun** [comment]
Please make this a subproject of `packages/angular` instead so we don't have another Angular project.

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
       "@clr/core/lit-html",
       "@clr/core/@storybook/**",
       "@clr/core/normalize.css",
+      "**/@clr/icons",
```
Comment (gnomeontherun) [inline package.json]: Why this?

File: packages/angular/angular.json [hunk 2]
```diff
@@ -269,6 +269,118 @@
...
+        }
+      },
+      "root": "projects/clarity-adoption",
+      "sourceRoot": "projects/clarity-adoption/src",
+      "prefix": "app",
+      "architect": {
+        "build": {
+          "builder": "@angular-devkit/build-angular:browser",
+          "options": {
+            "outputPath": "dist/clarity-adoption",
```
Comment (gnomeontherun) [inline packages/angular/angular.json]: This should be `"outputPath": "../../dist/website/adoption-guide",` so it outputs into the website project.

File: packages/angular/angular.json [hunk 3]
```diff
@@ -269,6 +269,118 @@
...
+          "options": {
+            "browserTarget": "clarity-adoption:build"
+          },
+          "configurations": {
+            "production": {
+              "browserTarget": "clarity-adoption:build:production"
+            }
+          }
+        },
+        "extract-i18n": {
```
Comment (gnomeontherun) [inline packages/angular/angular.json]: We can remove some of these blocks like extract, test, e2e.

File: yarn.lock [hunk 4]
```diff
@@ -8,18 +8,18 @@
   integrity sha512-RiMAVpSluRbWb2hlT9wMJ0r2l+MUZzScYjY+w2iWRzjOr9Zzzs4tYzJT6Sd94PDz3LzCuf4aGOwS6pkKXTEBLw==
 
 "@analytics/storage-utils@^0.2.2":
-  version "0.2.3"
```
Comment (gnomeontherun) [inline yarn.lock]: Can't we revert this file?

File: packages/eslint-plugin-clarity-adoption/src/rules/no-clr-button/disallowed-classes.ts [hunk 5]
```diff
@@ -1,6 +1,21 @@
 export const primaryDisallowedClass = 'btn';
 
-export const additionalDisallowedClasses = ['btn-primary', 'btn-success', 'btn-warning', 'btn-danger'];
+export const additionalDisallowedClasses = [
```
Comment (gnomeontherun) [inline packages/eslint-plugin-clarity-adoption/src/rules/no-clr-button/disallowed-classes.ts]: Why does this PR have changes to the eslint plugin together?

File: packages/angular/projects/clarity-adoption/src/index.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>Clarity Adoption</title>
+    <base href="/" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
```
Comment (gnomeontherun) [inline packages/angular/projects/clarity-adoption/src/index.html:8]: Can we make the Favicon Clarity logo?

File: package.json (hunk 1 above)
Comment (bdryanovski) [reply package.json]: This was for forcing yarn to install `@clr/icons` into apps now is no longer needed - is part of my next commit anyway

File: yarn.lock (hunk 4 above)
Comment (bdryanovski) [reply yarn.lock]: Yep I was rebuilding it as well - just merged with `next` so new angular is comming here as well

File: packages/eslint-plugin-clarity-adoption/src/rules/no-clr-button/disallowed-classes.ts (hunk 5 above)
Comment (bdryanovski) [reply packages/eslint-plugin-clarity-adoption/src/rules/no-clr-button/disallowed-classes.ts]: No idea maybe is something for @sis0k0

**Comment by bdryanovski** [comment]
Not sure why the `website` workflow fails, can't reproduce it locally ?! Need to investigate it a bit more...

**Review by bbogdanov — APPROVED** [review]

**Comment by bbogdanov** [comment]
@bdryanovski Do not forget to squash before merging!

**Review by sis0k0 — APPROVED** [review]

**Comment by bdryanovski** [comment]
I'm merging it because it's getting pretty hard to sync with all the changes lately. 

The thing that needs to be done so this to finish is:
  * Updating the landing page and maybe proof read it again.
  * Add as part of the website as an external link.

**Pull request merged**